packages/react-dom/src/__tests__/ReactMultiChild-test.js

```diff
@@ -323,6 +323,29 @@ describe('ReactMultiChild', () => {
     ReactDOM.render(<Foo />, div);
   });
 
+  it('should warn when using one-shot iterators as children', () => {
+    function Foo() {
+      const iterator = [<h1 key="1">Hello</h1>, <h1 key="2">Dave</h1>].values();
+
+      return iterator;
+    }
+
+    const div = document.createElement('div');
+    expect(() => {
+      ReactDOM.render(<Foo />, div);
+    }).toErrorDev(
+      'Using a TOOD-understandable-term as children is unsupported and will likely yield ' +
```

Comment:
Tried googling "one-shot iterator" and could only find mention of it related to tensorflow. Is this really a proper term in JS-land? Even if, we probably want to add an example "Using a one-shot iterator (e.g. Array.prototype.values()) as children [...]".

Reply:
Maybe just `iterator`?

> Using an iterator such as `[].values()` as children is unsupported and will likely yield unexpected results because it can only be iterated once. You may convert it [...]